packages/std/src/imports.rs

```diff
@@ -56,35 +57,33 @@ impl ExternalStorage {
         ExternalStorage {}
     }
 
-    pub fn get_with_result_length(&self, key: &[u8], result_length: usize) -> Option<Vec<u8>> {
+    pub fn get_with_result_length(
+        &self,
+        key: &[u8],
+        result_length: usize,
+    ) -> Result<Option<Vec<u8>>> {
         let key = build_region(key);
         let key_ptr = &*key as *const Region as *const c_void;
         let value_ptr = alloc(result_length);
 
         let read = unsafe { db_read(key_ptr, value_ptr) };
         if read == -1000002 {
-            panic!("Allocated memory too small to hold the database value for the given key. \
+            return contract_err("Allocated memory too small to hold the database value for the given key. \
```

Comment:
This is definitely nicer for error reporting (than some wasmer - "contract crashed" message)